server/src/main/java/org/opensearch/rest/RestController.java

```diff
@@ -505,6 +515,38 @@ private void handleBadRequest(String uri, RestRequest.Method method, RestChannel
         }
     }
 
+    /**
+     * Attempts to extracts auth token and login.
+     * 
+     * @returns false if there was an error and the request should not continue being dispatched
+     * */
+    private boolean handleAuthenticateUser(final RestRequest request, final RestChannel channel) {    
+        try {
+            final AuthToken token = RestTokenExtractor.extractToken(request);
+            // If no token was found, continue executing the request
+            if (token == null) {
+                return true;
+            }
```

Comment:
What should happen if identity is enabled but the auth token passed is `null`? My understanding is that the request should be blocked.

Reply:
The NoopSubject.java handles this case, no exceptions are thrown by design

```
    @Override
    public void login(AuthToken AuthToken) {
        // Do nothing as noop subject is always logged in
    }
```